graylog2-shared/src/main/java/org/graylog2/shared/plugins/PluginLoader.java

```diff
@@ -39,12 +48,17 @@ public PluginLoader(File pluginDir) {
     }
 
     public Set<Plugin> loadPlugins() {
-        final ImmutableSet.Builder<Plugin> plugins = ImmutableSet.builder();
+        final Set<Plugin> sorted = ImmutableSortedSet.orderedBy(new PluginComparator())
+                .addAll(loadClassPathPlugins())
+                .addAll(loadJarPlugins())
+                .build();
```

Comment:
As a matter of fact, the plugins in the classpath will also be contained in the collection returned by `loadJarPlugins()` because the same classloader is used as parent classloader for the `URLClassLoader` instances created in that method.